retrofit-cache/src/main/java/io/sam43/retrofitcache/interceptor/CacheInterceptor.kt

```
@@ -0,0 +1,192 @@
+            // Add request body to key calculation
+            val buffer = okio.Buffer()
+            request.body?.writeTo(buffer)
+            keyString += "_${buffer.readByteString().md5().hex()}"
+        }
+
+        // Use MD5 hash to create a shorter, consistent key
+        return try {
+            val md = MessageDigest.getInstance("MD5")
+            val digest = md.digest(keyString.toByteArray())
+            digest.joinToString("") { "%02x".format(it) }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)